doc/modules/multiclass.rst

```diff
@@ -14,52 +14,116 @@ Multiclass and multilabel algorithms
-  generalization of the multi-class classification task.  *The output format
-  is a 2d numpy array or sparse matrix.*
-
-  The set of labels can be different for each output variable.
-  For instance, a sample could be assigned "pear" for an output variable that
-  takes possible values in a finite set of species such as "pear", "apple"; 
-  and "blue" or "green" for a second output variable that takes possible values
-  in a finite set of colors such as "green", "red", "blue", "yellow"...
-
-  This means that any classifiers handling multi-output
-  multiclass or multi-task classification tasks,
-  support the multi-label classification task as a special case.
-  Multi-task classification is similar to the multi-output
-  classification task with different model formulations. For
-  more information, see the relevant estimator documentation.
+- **Multiclass classification**: classification task which labels each
+  sample as **one** of a number of classes, where the number of classes is
+  greater than 2. Each sample can only be labelled as one class.
+
+  For example, classification using features extracted from a set of images of
+  fruit, where each image may either be of an orange, an apples, or a pear.
+  Each image is one sample and is labelled as one of the 3 possible classes.
+  Multiclass classification makes the assumption that each sample is assigned
+  to one and only one label - one sample cannot, for example, be both a pear
+  and an apple at the same time.
+  
+  - Valid :term:`multiclass` representations for
+    :func:`~utils.multiclass.type_of_target` (`y`) are:
+
+    - 1d or column vector containing more than two discrete values.
+    - sparse :term:`binary` matrix of shape ``(n_samples, n_classes)`` with a
+      single element per row, where each column represents one class.
+
+
+- **Multilabel classification**: classification task which labels each sample
+  with a set of binary attributes (either True or False) independently of
+  one another. The number of binary attributes is greater or equal to 2.
+
+  For example, classification of 2 fruit attributes, crispy or not crispy and
+  sour or not sour, using features extracted from a set of images of fruit.
+  Each sample is an image of one fruit, labels for both binary attributes are
+  output for each sample and each attribute is labelled independently of the
+  other attribute.
+
+  - valid representation :term:`multilabel` `y` is:
+
+    - either dense (or sparse) :term:`binary` matrix of shape
+      ``(n_samples, n_classes)`` with multiple active elements per row to denote
+      that the sample belongs to multiple classes. Each column represents a class.
+
+
+- **Multioutput regression**: predicts multiple numerical properties for each
+  sample. Each property is a continuous variable and the number of properties
+  to be predicted for each sample is greater or equal to 2.
+
+  For example, prediction of weight, sugar content and number of calories using
```

Comment:
I'm not sure predicting weight, sugar content and number of calories from an image makes much sense here. Maybe just keep the previous example

Reply:
The old example was: such as wind direction and magnitude at a certain location - but I am not sure if wind direction is numerical?

I have amended it to predicting surface area and volume, so it hopefully makes more sense.